htdocs/core/lib/modulebuilder.lib.php

```diff
@@ -416,3 +416,70 @@ function rebuildObjectSql($destdir, $module, $objectname, $newmask, $readdir = '
 
 	return $error ? -1 : 1;
 }
+
+/**
+ * Rewriting all permissions after updating file
+ * @param string         $file         file with path
+ * @param string         $objectname   name of the permission object
+ * @param string         $right        permission of object
+ * @param string         $rightUpdated permission of object updated
+ * @param int            $action       0 for delete, 1 for add, 2 fo update
+ * @return void
+ */
+function rewritingAllPermissions($file, $objectname, $right, $rightUpdated, $action)
+{
+	global $langs;
+	$start = "/* BEGIN MODULEBUILDER PERMISSIONS */";
+	$end = "/* END MODULEBUILDER PERMISSIONS */";
+	$perms_after_action = '';
+	$error = 0;
+	if (empty($right)) {
+		$error++;
+	}
+	if ($action == 2 && empty($rightUpdated)) {
+		$error++;
+	}
+	if (!$error) {
+		// Open the file and read line by line
+		$handle = fopen($file, "r");
```

Comment:
Here we prefer to use a file_get_contents() for security purpose

Reply:
ok , i see